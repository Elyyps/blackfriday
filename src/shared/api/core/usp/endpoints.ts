import { uniqueSellingPoints } from "./generate-dummy-data";
import { USP } from "./usp";

export const generateUSP = (): USP => uniqueSellingPoints;
