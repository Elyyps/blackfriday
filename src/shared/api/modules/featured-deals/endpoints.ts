import { dummyFeaturedDealsData } from "./generate-dummy-data";

export const getFeaturedDealsDummyData = () => dummyFeaturedDealsData;
