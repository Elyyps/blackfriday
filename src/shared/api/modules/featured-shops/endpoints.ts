import { dummyFeaturedShopsData } from "./generate-dummy-data";

export const getFeaturedShopsDummyData = () => dummyFeaturedShopsData;
