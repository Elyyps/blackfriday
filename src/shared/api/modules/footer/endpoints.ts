import { bottomFooter, middleFooter, topFooter } from "./generate-dummy-data";
import { Footer } from "./models/footer.module";

export const getFullfooter = (): Footer => ({
  bottomFooter: bottomFooter(),
  middleFooter: middleFooter(),
  topFooter: topFooter()
});
