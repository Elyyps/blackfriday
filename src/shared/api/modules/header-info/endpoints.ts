import { HeaderInfoDummy } from "./generate-dummy-data";

export const getHeaderInfoDummyData = () => HeaderInfoDummy;
