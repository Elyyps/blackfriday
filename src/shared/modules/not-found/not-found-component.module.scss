.not-found {
  background-color: $gray-dark;
  z-index: 0;
  height: 91vh;
  display: flex;
  align-items: center;
  justify-content: center;

  &-container {
    margin-top: -40px;
    align-items: center;
    flex-direction: column;
    color: white;
    display: flex;
  }
  &:after {
    content: "";
    position: absolute;
    top: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-image: url("../../../assets/pattern.svg");
    background-position: 50% 0;
    background-size: 2000px auto;
    opacity: 0.5;
  }

  &-title {
    padding: 30px 0 0 0;
    font-family: "arial";
    font-size: 48px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    @media (max-width: 600px) {
      font-size: 40px;
    }
  }
  a {
    margin-top: 25px;
    width: 204px;
    height: 36px;
    border-radius: 2px;
    @media (max-width: 600px) {
      width: 190px;
      height: 33px;
      font-size: 13px;
      margin-top: 20px;
    }
  }
  p {
    font-family: "arial";
    font-size: 16px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    text-align: center;
    @media (max-width: 600px) {
      font-size: 12px;
    }
    @media (max-width: 420px) {
      padding: 0 5% 0 5%;
      white-space: pre-wrap;
    }
  }
}
