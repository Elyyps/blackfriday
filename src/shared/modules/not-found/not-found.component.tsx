import * as React from "react";
import styles from "./not-found-component.module.scss";
import { NotFoundModule } from "@app/api/modules/not-found/not-found";
import { Button, IconComponent } from "@app/core";
import RoadSignRight from "@assets/icons/road-sign-right.svg";
export interface INotFoundComponentProps {
  notFoundModule: NotFoundModule;
}

const NotFoundComponent = (props: INotFoundComponentProps) => (
  <div className={styles["not-found"]}>
    <div className={styles["not-found-container"]}>
      <IconComponent icon={RoadSignRight} size={"45px"} />
      <span className={styles["not-found-title"]}>{props.notFoundModule.tilte}</span>
      <p>{props.notFoundModule.content}</p>
      <Button title={props.notFoundModule.button.title} variant="secondary" />
    </div>
  </div>
);

export { NotFoundComponent };
