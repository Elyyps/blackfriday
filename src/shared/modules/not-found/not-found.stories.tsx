import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { NotFoundComponent } from "./not-found.component";
import { generateNotFoundModuleDummyData } from "@app/api/modules/not-found/endpoints";
const data = generateNotFoundModuleDummyData();
storiesOf("NotFound", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <NotFoundComponent notFoundModule={data} />);
