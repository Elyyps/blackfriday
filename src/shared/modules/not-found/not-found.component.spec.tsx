import React from "react";
import { shallow } from "enzyme";
import { NotFoundComponent } from "./not-found.component";
import { generateNotFoundModuleDummyData } from "@app/api/modules/not-found/endpoints";
const data = generateNotFoundModuleDummyData();
describe("[NotFound]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<NotFoundComponent notFoundModule={data} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
