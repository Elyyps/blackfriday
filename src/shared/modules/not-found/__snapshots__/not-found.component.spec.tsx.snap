// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[NotFound] should render component without crashing 1`] = `
<div
  className="not-found"
>
  <div
    className="not-found-container"
  >
    <IconComponent
      icon="road-sign-right.svg"
      size="45px"
    />
    <span
      className="not-found-title"
    >
      Oeps!
    </span>
    <p>
      We hebben de pagina niet gevonden. Zullen we je de weg wijzen?
    </p>
    <ClickableComponent
      href="./"
      title="Black Friday homepage"
      variant="secondary"
    />
  </div>
</div>
`;
